pkgs/servers/klipper/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, python2
+}:
+stdenv.mkDerivation rec {
+  name = "klipper";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "KevinOConnor";
+    repo = "klipper";
+    rev = "v${version}";
+    sha256 = "1ijy2ij9yii5hms10914i614wkjpsy0k4rbgnm6l594gphivdfm7";
+  };
+
+  sourceRoot = "source/klippy";
```

Comment:
This looks like it is not klipper, what is packaged, but rather just one of its components klippy?

Reply:
Klipper is two projects in one. There is the klipper firmware and the klipper daemon. The firmware is unlikely to ever end up nixpkgs as it spits out a `.bin` that you then manually either flash to your board, or put in an SD card with a certain name, depending on the board. The daemon is what _is_ being packaged here and does much of the 3D printer computations and talks to the firmware on the controller. 

It was suggested on IRC to call this `klipper-klippy` but I don't see us packaging the firmware, and so don't see the point. For reference, I did go through the trouble of making a Nix drv that builds the firmware https://github.com/lovesegfault/voronix/blob/master/klipper.nix, I just don't think it belongs in nixpkgs.